### PR TITLE
reflect status of job/service/ingress to resource template

### PR DESCRIPTION
**Pull request by mrlihanbo**
Signed-off-by: lihanbo <lihanbo2@huawei.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
reflect status of job/service/ingress to resource template

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
aggregate status of load balance service:
```
spec:
  clusterIP: 10.100.245.74
  externalTrafficPolicy: Cluster
  ports:
  - nodePort: 31963
    port: 80
    protocol: TCP
    targetPort: 8080
  selector:
    app: nginx
  sessionAffinity: None
  type: LoadBalancer
status:
  loadBalancer:
    ingress:
    - hostname: member1
      ip: 192.0.2.127
```

aggregate status of job"
```status:
    conditions:
  - lastProbeTime: "2021-08-18T09:59:02Z"
    lastTransitionTime: "2021-08-18T09:59:02Z"
    message: Job executed failed in member clusters member3,member1
    reason: JobFailed
    status: "True"
    type: Failed
  failed: 16
```

```status:
  conditions:
  - lastProbeTime: "2021-08-16T11:27:01Z"
    lastTransitionTime: "2021-08-16T11:27:01Z"
    message: Job completed
    reason: Completed
    status: "True"
    type: Complete
  succeeded: 12

```

**Does this PR introduce a user-facing change?**:
“NONE”

### CODE REVIEW

**Comment by RainbowMango** [comment]
You might want to add an E2E test for each of them.

**Comment by mrlihanbo** [comment]
> You might want to add an E2E test for each of them.

I can add e2e of agregating job status and ingress propagation, but I can't add the e2e of agregating service and ingress status due to kind clusters do not have lb controller and ingress controller.

File: pkg/util/detector/aggregate_status.go [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+		}
+		klog.Errorf("Failed to get service(%s/%s): %v", objRef.Namespace, objRef.Name, err)
+		return err
+	}
+
+	// if job was finished previously, we don't need to redo the aggregate
+	aggregated, _ := d.getJobFinishedStatus(&obj.Status)
+	if aggregated {
+		return nil
+	}
```
Comment (RainbowMango) [inline pkg/util/detector/aggregate_status.go:210]:
```suggestion
```
This will block us from updating the status.

**Review by RainbowMango — CHANGES_REQUESTED** [review]

**Pull request merged**